src/Parser/Core/Modules/Items/Legion/EyeOfCommand.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import { calculateSecondaryStatDefault } from 'common/stats';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import { formatPercentage } from 'common/format';
+
+/**
+ * Eye of Command
+ * Your melee auto attacks increase your Critical Strike by X for 10 sec, stacking up to 10 times. This effect is reset if you auto attack a different target.
+ */
+class EyeOfCommand extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  [SPELLS.LEGIONS_GAZE.id] = {
+    1: {
+      start: 0,
+      uptime: 0,
+    },
+    2: {
+      start: 0,
+      uptime: 0,
+    },
+    3: {
+      start: 0,
+      uptime: 0,
+    },
+    4: {
+      start: 0,
+      uptime: 0,
+    },
+    5: {
+      start: 0,
+      uptime: 0,
+    },
+    6: {
+      start: 0,
+      uptime: 0,
+    },
+    7: {
+      start: 0,
+      uptime: 0,
+    },
+    8: {
+      start: 0,
+      uptime: 0,
+    },
+    9: {
+      start: 0,
+      uptime: 0,
+    },
+    10: {
+      start: 0,
+      uptime: 0,
+    },
+  };
+
+  critPerStack = 0;
+  currentStacks = 0;
+  buffIsUp = false;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTrinket(ITEMS.EYE_OF_COMMAND.id);
+    if (this.active) {
+      const itemDetails = this.combatants.selected.getItem(ITEMS.EYE_OF_COMMAND.id);
+      this.critPerStack = calculateSecondaryStatDefault(865, 151, itemDetails.itemLevel);
+    }
+  }
+
+  on_byPlayer_applybuff(event) {
```

Comment:
Have you looked at the `changebuffstack` fabricated event? I think you should be able to reduce the total amount of code as well as the amount of duplicate code with that.

Reply:
The problem is that the buffs don't clear on ending combat, leaving me with inaccurate numbers if I was to simply adopt the Garothi Feedback direction.

Reply:
I suppose that's a bug in GFC, I'll see if I can fix